cmd/integrationArtifactGetMplStatus.go

```diff
@@ -0,0 +1,112 @@
+	// i.e. include everything you need to be able to mock in tests.
+	// Unit tests shall be executable in parallel (not depend on global state), and don't (re-)test dependencies.
+}
+
+type integrationArtifactGetMplStatusUtilsBundle struct {
+	*command.Command
+
+	// Embed more structs as necessary to implement methods or interfaces you add to integrationArtifactGetMplStatusUtils.
+	// Structs embedded in this way must each have a unique set of methods attached.
+	// If there is no struct which implements the method you need, attach the method to
+	// integrationArtifactGetMplStatusUtilsBundle and forward to the implementation of the dependency.
+}
+
+func newIntegrationArtifactGetMplStatusUtils() integrationArtifactGetMplStatusUtils {
+	utils := integrationArtifactGetMplStatusUtilsBundle{
+		Command: &command.Command{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func integrationArtifactGetMplStatus(config integrationArtifactGetMplStatusOptions, telemetryData *telemetry.CustomData, commonPipelineEnvironment *integrationArtifactGetMplStatusCommonPipelineEnvironment) {
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
+	httpClient := &piperhttp.Client{}
+	// For HTTP calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runIntegrationArtifactGetMplStatus(&config, telemetryData, httpClient, commonPipelineEnvironment)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runIntegrationArtifactGetMplStatus(config *integrationArtifactGetMplStatusOptions, telemetryData *telemetry.CustomData, httpClient piperhttp.Sender, commonPipelineEnvironment *integrationArtifactGetMplStatusCommonPipelineEnvironment) error {
+	clientOptions := piperhttp.ClientOptions{}
+	httpClient.SetOptions(clientOptions)
+	header := make(http.Header)
+	header.Add("Accept", "application/json")
+
+	mplStatusURL := fmt.Sprintf("%s/api/v1/MessageProcessingLogs?$filter=IntegrationArtifact/Id eq '%s'&$orderby=LogEnd desc&$top=1", config.Host, config.IntegrationFlowID)
+	tokenParameters := cpi.TokenParameters{TokenURL: config.OAuthTokenProviderURL, Username: config.Username, Password: config.Password, Client: httpClient}
+	token, err := cpi.CommonUtils.GetBearerToken(tokenParameters)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch Bearer Token")
+	}
+	clientOptions.Token = fmt.Sprintf("Bearer %s", token)
+	httpClient.SetOptions(clientOptions)
+	mplStatusEncodedURL := strings.ReplaceAll(mplStatusURL, " ", "%20")
```

Comment:
I would expect that url escaping is performed centrally inside the http utils. I think it is wrong to url-escape here. And in case I'm wrong and it needs to be done here: we should not only handle the space (`%20`) but call some url-escaping facility escaping everything which needs to be escaped - basically something like [here](https://golang.org/pkg/net/url/#QueryEscape). But again: I believe this should be done centrally inside the http  tooling.